### DISCUSSION TITLE
Not populated Command Palette on Linux (?)

### DISCUSSION THREAD

**Discussion by kuuff** [zed-industries/zed#14394] [Feature Requests]
Hi. I think Zed doesn't work properly for me on Linux, but I'm not entirely sure for which extent. For example, it suggests to use Ctrl-O to open a project, I press Ctrl-O and nothing happens. The only way I've found to open files is to run zed from command line like `zed file-to-open`.

I believe it goes deeper then just Ctrl-O, because when I tried to do the exercises on how to use Zed editor, I was unable to do anything, the commands mentioned in the tutorial are missing from Command Palette. I type them in the search bar and they are not found. 

Some features work though. I can edit text files. I can type text in, delete, move cursor around with arrow keys or mouse. Embedded terminal works great. If I run Zed with `zed directory-to-open` then it opens the directory as a project allowing me to navigate files. Then I can reopen this project through `open recent project`. If I open sources in Rust, Zed installs rust-analyzer and it works also.

The trouble is I'm not sure if it due to Zed is not completely ready for Linux and what I see is what I should see, or maybe something is wrong on my system? Because the most probable culprit for this issues is my system lacking some dependencies or something. I use gentoo, not anything popular like Ubuntu, and my other software choices often are not common also. Like Fluxbox for a window manager, not KDE or Gnome, and no systemd, I'm totally happy with OpenRC. The trouble is I can't see how my software choices could prevent Zed to show me a dialog for choosing a file, or to populate Command Palette with commands. So I don't know really where to look for what is wrong with it, I'm even not sure there is something wrong.

BTW I tried to build Zed from github sources, just now I tried the install script that downloads ready-made binaries, to no avail.

**Comment by 0xDEAD-C0DE** [comment]
Do you have a xdg-desktop portal installed?  What happened when you build from source? How about when you used the `install.sh` script? Have you tried this? https://zed.dev/docs/linux#other-ways-to-install-zed-on-linux

**Comment by kuuff** [comment]
> Do you have a xdg-desktop portal installed?

No, I don't. I'll try to install it later. (world is being built now, and looking at what emerge says to me it is just the preparation stage for the full system rebuilding, so hopefully I'll be able to try installing xdg tomorrow)

> What happened when you build from source?

I tried it with `cargo build --release`, the build was successful, and Zed started, but something was wrong. I don't remember clearly what exactly was wrong, but I believe it was like I described above.

Then I went to read README.md, docs on how to install on different platforms and so on, and I discovered `./script/install-linux` and used it. The result was how I described above.

> Have you tried this? https://zed.dev/docs/linux#other-ways-to-install-zed-on-linux

I didn't tried to use package manager, because Gentoo doesn't work that way. Either the package in portage, or there is no package.

But anyway, xdg looks promising. I can imagine ways how the lack of xdg could lead to these symptoms.

Thank you.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
